src/vs/workbench/contrib/notebook/browser/notebookServiceImpl.ts

```diff
@@ -153,15 +162,57 @@ export class NotebookProviderInfoStore extends Disposable {
 
 	private _convertPriority(priority?: string) {
 		if (!priority) {
-			return NotebookEditorPriority.default;
+			return ContributedEditorPriority.default;
 		}
 
 		if (priority === NotebookEditorPriority.default) {
-			return NotebookEditorPriority.default;
+			return ContributedEditorPriority.default;
 		}
 
-		return NotebookEditorPriority.option;
+		return ContributedEditorPriority.option;
+
+	}
+
+	private registerContributionPoint(notebookProviderInfo: NotebookProviderInfo): void {
+		for (const selector of notebookProviderInfo.selectors) {
+			const globPattern = (selector as INotebookExclusiveDocumentFilter).include || (selector as glob.IRelativePattern).pattern || selector;
+			this._register(this._extensionContributedEditorService.registerContributionPoint(
```

Comment:
Pretty sure this shouldn't be registered globally. When the extension host restarts we re-run this code and with the global disposals we don't clean-up registerations from the previous extension host

Reply:
Maybe it does work. This needs testing or a second pair of :eyes: from @rebornix. The fact that there is `_handled` is fishy

Reply:
I'm unsure of the lifecycle of this piece of code, is there some "bad" signs I should be looking for?

Reply:
F1 > Restart Extension Host will tell you. You this is invoked again without having remove the previous contribution (L179) than it is a problem

Reply:
I don't see this getting invoked, I also don't really see anything happen when I restart the extension host.

Reply:
Sorry, gave the wrong hint. This happens when disabling/enabling extensions. I still believe that dispoables pile up in that case

Reply:
Ah yes, this does happen.